prov/rxm/src/rxm_cq.c

```diff
@@ -1217,17 +1218,19 @@ static void rxm_cq_read_write_error(struct rxm_ep *rxm_ep)
 
 static inline int rxm_ep_repost_buf(struct rxm_rx_buf *rx_buf)
 {
+	int ret;
+
 	if (rx_buf->ep->srx_ctx)
 		rx_buf->conn = NULL;
 	rx_buf->hdr.state = RXM_RX;
 
-	if (fi_recv(rx_buf->msg_ep, &rx_buf->pkt,
-		    rx_buf->ep->eager_limit + sizeof(struct rxm_pkt),
-		    rx_buf->hdr.desc, FI_ADDR_UNSPEC, rx_buf)) {
-		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL, "Unable to repost buf\n");
-		return -FI_EAVAIL;
-	}
-	return FI_SUCCESS;
+	ret = fi_recv(rx_buf->msg_ep, &rx_buf->pkt,
+		      rx_buf->ep->eager_limit + sizeof(struct rxm_pkt),
+		      rx_buf->hdr.desc, FI_ADDR_UNSPEC, rx_buf);
+	if (ret)
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+			"Unable to repost buf: %d\n", ret);
+	return ret;
```

Comment:
This function is converting ssize_t (from fi_recv) into an int (ret).  I don't know if you want to convert rxm_ep_repost_buf from int to ssize_t, or just apply a cast.  If you go with the first option, then series is okay to merge, with a follow up cleanup patch at some point.

Reply:
Sure. I'll convert int to ssize_t in a follow-up patch.